src/org.ts

```diff
@@ -211,6 +211,49 @@ export class Org extends AsyncCreatable<Org.Options> {
     }
   }
 
+  /**
+   * If this Org is a scratch org, calling this method will delete the scratch org from the DevHub and clean up any local files
+   *
+   * @param hubOrg - optional DevHub Org of the to-be-deleted scratch org
+   */
+  public async deleteScratchOrg(hubOrg?: Org): Promise<void> {
+    const logger = await Logger.child('deleteScratchOrg');
+
+    // if we didn't get a hubOrg, we'll get it from the this org
+    hubOrg = hubOrg ?? (await this.getDevHubOrg());
+    if (hubOrg?.getOrgId() === this.getOrgId()) {
+      // we're attempting to delete a DevHub
+      throw SfdxError.create('@salesforce/core', 'org', 'deleteOrgHubError');
+    }
+    if (hubOrg) {
+      const hubOrgConn = hubOrg.getConnection();
+
+      try {
+        const activeScratchOrgRecordId = (
+          await hubOrgConn.singleRecordQuery<{ Id: string }>(
+            `SELECT Id FROM ActiveScratchOrg WHERE SignupUsername='${this.getUsername()}'`
+          )
+        ).Id;
+        logger.trace(`found matching ActiveScratchOrg with SignupUsername: ${this.getUsername()}.  Deleting...`);
+        await hubOrgConn.delete('ActiveScratchOrg', activeScratchOrgRecordId);
```

Comment:
can these happen in parallel?

Reply:
I don't think so because the `hubOrgConn.delete` could fail, and the local `this.remove` would clear auth files so your scratch org would be in a remote only state without an easy way (from the CLI) to access it right?

thinking about this some more, that above scenario might be fine. It might be a question of behavior :hmm: 🤔